**(b) Modelo de Programación Lineal para Encontrar una Política Óptima**

In [3]:
from scipy.optimize import linprog

# Coeficientes de la función objetivo.
#     y01 y02 y11 y12 y21 y22
obj = [3,  9,  3,  9,  28, 34]

# Lado izquierdo de las restricciones de igualdad.
lhs_eq = [[1, 1, 1, 1, 1, 1],
          [1/2, 1/2, -3/10, -2/5, 0, 0],
          [-1/2, -1/2, 1/2, 1/2, -3/5, -4/5],
          [0, 0, -2/10, -1/10, 3/5, 4/5]]

# Lado derecho de las restricciones de igualdad.
rhs_eq = [1, 0, 0, 0]

# Posibles valores de las variables de restricción. Mayor o igual a 0.
bnd = [(0, float("inf")),
       (0, float("inf")),
       (0, float("inf")),
       (0, float("inf")),
       (0, float("inf")),
       (0, float("inf"))]

# Resolver el modelo de programación lineal.
opt = linprog(c=obj, A_eq=lhs_eq, b_eq=rhs_eq, bounds=bnd, method="revised simplex")

# Mostrar el resultado.
print(opt.x)

[0.32432432 0.         0.54054054 0.         0.         0.13513514]


/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:21: OptimizeWarning: A_eq does not appear to be of full row rank. To improve performance, check the problem formulation for redundant equality constraints.


Según los resultados obtenidos:

$y_{01} = 0.32432432$

$y_{02} = 0$

$y_{11} = 0.54054$

$y_{12} = 0$

$y_{21} = 0$

$y_{22} = 0.13513514$

Por lo tanto, la **política óptima** es tomar la decisión 1 en los estados 0 y 1 y tomar la decisión 2 en el estado 2. 